fix: quests from version 10 revised

htc16 opened the pull request:
# Description

- [x] 10.10 - [Fury Gates Mini World Change](https://tibia.fandom.com/wiki/Fury_Gates_Mini_World_Change) (**need to develop this quest**)
- [x] 10.10 - [Hunting for Tokens Quest](https://tibia.fandom.com/wiki/Hunting_for_Tokens_Quest) (**need to develop this quest**)
- [x] 10.10 - [Mysterious Ornate Chest Quest](https://tibia.fandom.com/wiki/Mysterious_Ornate_Chest)
- [x] 10.10 - [Small Sapphire Quest](https://tibia.fandom.com/wiki/Small_Sapphire_Quest)
- [x] 10.10 - [The Gravedigger of Drefia Quest](https://tibia.fandom.com/wiki/The_Gravedigger_of_Drefia_Quest)

In this position (33025, 32331, 10) all items ID 18617 need to be replaced with ID 32688 and in positions (33025, 32332, 10) and (33026, 32332, 10) they also need to have ID **32688 on the bottom of the stone table.**

- [x] 10.10 - [The Repenters Quest](https://tibia.fandom.com/wiki/The_Repenters_Quest) (**need to develop this quest**)
- [x] 10.20 - [Cave Explorer Outfits Quest](https://tibia.fandom.com/wiki/Cave_Explorer_Outfits_Quest)
- [x] 10.20 - [Spike Tasks Quest](https://tibia.fandom.com/wiki/Spike_Tasks_Quest)

Position: (32277, 32679, 8) the top map gets in the way of the bottom one.
At Position: {x = 32293, y = 32539, z = 9} there is a missing tile.

- [x] 10.20 - [Spirit Grounds Mini World Change](https://tibia.fandom.com/wiki/Spirit_Grounds_Mini_World_Change) (**need to develop this quest**)
- [x] 10.30 - [Dream Warden Outfits Quest](https://tibia.fandom.com/wiki/Dream_Warden_Outfits_Quest) (**need to develop this quest**)
- [x] 10.30 - [Roshamuul Quest](https://tibia.fandom.com/wiki/Roshamuul_Quest) (**need to develop this quest**)
- [x] 10.32 - [Tutorial](https://tibia.fandom.com/wiki/Tutorial) Dawnport
- [x] 10.37 - [Chyllfroest Mini World Change](https://tibia.fandom.com/wiki/Chyllfroest_Mini_World_Change) (**need to develop this quest**)
- [x] 10.37 - [Tinder Box Quest](https://tibia.fandom.com/wiki/Tinder_Box_Quest)

At position { x = 32085, y = 31025, z = 1 } place a body id 4285
delete the corpse at Position: (32081, 31028, 2) as it is in the wrong place.

- [x] 10.50 - [Dark Trails Quest](https://tibia.fandom.com/wiki/Dark_Trails_Quest)

At position (33480, 32085, 9), (33481, 32085, 9) and (33481, 32086, 9) there are items that need to be removed, they are: 03 glooth amulet and 01 glooth axe.
And at position (33488, 32090, 9) change the teleport position to position (33584, 31847, 10).

- [x] 10.50 - [Glooth Engineer Outfits Quest](https://tibia.fandom.com/wiki/Glooth_Engineer_Outfits_Quest)
- [x] 10.50 - [Oramond Quest](https://tibia.fandom.com/wiki/Oramond_Quest)
- [x] 10.50 - [Rathleton Quest](https://tibia.fandom.com/wiki/Rathleton_Quest) (**need to develop this quest**)
- [x] 10.55 - [Dawnport Quest](https://tibia.fandom.com/wiki/Dawnport_Quest)
- [x] 10.55 - [Sanctuary of the Lizard God Quest](https://tibia.fandom.com/wiki/Sanctuary_of_the_Lizard_God_Quest) Dawnport
- [x] 10.70 - [Hero of Rathleton Quest](https://tibia.fandom.com/wiki/Hero_of_Rathleton_Quest) (**need to develop this quest**)
- [x] 10.70 - [Lion's Rock Quest](https://tibia.fandom.com/wiki/Lion%27s_Rock_Quest)
- [x] 10.70 - [The Cursed Crystal Quest](https://tibia.fandom.com/wiki/The_Cursed_Crystal_Quest)

at position (31974, 32907, 8) and (32031, 32914, 8) swap the skeleton body id for id 4285.
At position (31939, 32897, 6) add item id 2541.

- [x] 10.70 - [The Fire-Feathered Serpent World Change](https://tibia.fandom.com/wiki/The_Fire-Feathered_Serpent_World_Change) (**need to develop this quest**)
- [x] 10.70 - [Twenty Miles Beneath the Sea Quest](https://tibia.fandom.com/wiki/Twenty_Miles_Beneath_the_Sea_Quest) (**need to develop this quest**)
- [x] 10.80 - [Asura Palace Quest](https://tibia.fandom.com/wiki/Asura_Palace_Quest) (**need to develop this quest**)
- [x] 10.80 - [Cartography 101 Quest](https://tibia.fandom.com/wiki/Cartography_101_Quest) (**need to develop this quest**)
- [x] 10.80 - [Chakoya Iceberg Mini World Change](https://tibia.fandom.com/wiki/Chakoya_Iceberg_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [Grimvale Mini World Change](https://tibia.fandom.com/wiki/Grimvale_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [Grimvale Quest](https://tibia.fandom.com/wiki/Grimvale_Quest) (**need to develop this quest**)
- [x] 10.80 - [Hive Outpost Mini World Change](https://tibia.fandom.com/wiki/Hive_Outpost_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [Jungle Camp Mini World Change](https://tibia.fandom.com/wiki/Jungle_Camp_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [Nightmare Isles Mini World Change](https://tibia.fandom.com/wiki/Nightmare_Isles_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [Nightmare Teddy Quest](https://tibia.fandom.com/wiki/Nightmare_Teddy_Quest)
- [x] 10.80 - [Poacher Caves Mini World Change](https://tibia.fandom.com/wiki/Poacher_Caves_Mini_World_Change) (**need to develop this quest**)
- [x] 10.80 - [The Great Dragon Hunt Quest](https://tibia.fandom.com/wiki/The_Great_Dragon_Hunt_Quest)
- [x] 10.80 - [The Lost Brother Quest](https://tibia.fandom.com/wiki/The_Lost_Brother_Quest)
- [x] 10.80 - [The Tainted Souls Quest](https://tibia.fandom.com/wiki/The_Tainted_Souls_Quest) (**need to develop this quest**)
- [x] 10.90 - [Ferumbras' Ascension Quest](https://tibia.fandom.com/wiki/Ferumbras%27_Ascension_Quest)

At position {x = 33671, y = 32637, z = 13} update the map that is on the downloads page on Discord.
At position {x = 33427, y = 32852, z = 13} remove the actionid that is in the tile.
At position {x = 33466, y = 32392, z = 13} remove the actionid that is in the tile.
At position {x = 33204, y = 31510, z = 13} remove the actionid that is in the tile.
At position {x = 33357, y = 32440, z = 13} remove the actionid that is in the tile.
At position {x = 33678, y = 32758, z = 13} remove the actionid that is in the tile.
At position {x = 33614, y = 32679, z = 15} remove the actionid that is in the tile.
At position {x = 33459, y = 32820, z = 14} remove the actionid that is in the tile.

- [x] 10.90 - [Krailos Quest](https://tibia.fandom.com/wiki/Krailos_Quest) (**need to develop this quest**)
- [x] 10.90 - [Rift Warrior Outfits Quest](https://tibia.fandom.com/wiki/Rift_Warrior_Outfits_Quest)
- [x] 10.94 - [Heart of Destruction Quest](https://tibia.fandom.com/wiki/Heart_of_Destruction_Quest) (**need to develop this quest**)
- [x] Quests have been tested in development!


### _I'm currently working on adding all quests to the PR. Given the large number of quests, I would like to ask for your help and collaboration to close this PR as quickly as possible. If you are interested in helping, choose one of the quests to review, open a PR for it, and link to this PR. All help is welcome!_


Note1: my focus is on quests, therefore, the quest log for a quest may be missing as this is not shown in the quest tutorial lists. I study tutorials and release fix ones or develop something related to the tutorial.

Note2: I have limitations as this is the first time I am willing to do this, so please consider that some codes may not be the best but what matters is the intention and this can be improved in the future if someone finds it appropriate to do so.

Note3: I'm taking the opportunity to catalog all the aids and uids so that we can eventually modify and change them in the future directly through the script.

Note4: this PR needs some updates to the map that were reported above.

Note5: I was unable to place the writings in the books, only through the map editor, but this did not influence the development of the quest, so I left this for a future review.

## Behavior
### **Real**
I reviewed and corrected any quests that weren't working.

Delete options that are not relevant.

 - [x] Bug fix (rolling change that fixes a problem)
 - [x] New feature (rolling change that adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How this was tested

As I reviewed and modified the codes, I tested them.

**Test Setup**:

 - Server version: current
 - Client: 13
 - Operating system: Windows 11

## Checklist

 - [x] My code follows the style guidelines of this project
 - [x] I performed a self-review of my own code
 - [x] I checked the PR checks reports
 - [x] I commented my code, especially in areas that were difficult to understand
 - [x] I made corresponding changes to the documentation
 - [x] My changes do not generate new warnings
 - [x] I added tests that prove my fix is ​​effective or that my feature works

htc16 committed:
Hunting for Tokens Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Merge remote-tracking branch 'upstream/main' into Review-Quests-V10

htc16 committed:
The Gravedigger of Drefia Quest - part 1

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Gravedigger of Drefia Quest - last part

github-actions[bot] committed:
Lua code format - (Stylua)

omarcopires commented on data-otservbr-global/scripts/quests/the_gravedigger_of_drefia/movements_sacrifice_teleport.lua:
```diff
@@ -6,16 +6,17 @@ function sacrificeTeleport.onStepIn(creature, item, position, fromPosition)
 		return true
```
```lua
local teleportDestinations = {
	[4541] = {position = Position(33017, 32419, 11), storageKey = Storage.Quest.U10_10.TheGravediggerOfDrefia.Mission74},
	[4542] = {position = Position(33018, 32425, 11), storageKey = Storage.Quest.U10_10.TheGravediggerOfDrefia.Mission75}
}

local sacrificeTeleport = MoveEvent()

function sacrificeTeleport.onStepIn(creature, item, position, fromPosition)
	local player = creature:getPlayer()
	if not player then
		return true
	end

	local destination = teleportDestinations[item.actionid]
	if destination and player:getStorageValue(destination.storageKey) == 1 then
		player:teleportTo(destination.position)
		player:getPosition():sendMagicEffect(CONST_ME_TELEPORT)
	end

	return true
end

sacrificeTeleport:type("stepin")

for actionId in pairs(teleportDestinations) do
	sacrificeTeleport:aid(actionId)
end

sacrificeTeleport:register()
```

htc16 replied:
Thank you @omarcopires

htc16 committed:
Update movements_sacrifice_teleport.lua

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Spike Task Quest - part 1

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Spike Task Quests - Tasks 25-49

htc16 committed:
Spike Task Quests - Tasks 50-79

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Spike Task Quests - Tasks 80+

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Roshamuul Quest

I managed to make some scripts that weren't working work, however, I realized that all or part of the task mechanics needed to be developed.

htc16 committed:
Tinder Box Quest - Event Chyllfroest

htc16 committed:
Tinder Box Quest - Event Chyllfroest - fix

htc16 committed:
Dark Trails Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Glooth Engineer Outfits Quest

htc16 committed:
Oramond Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Sanctuary of the Lizard God Quest

htc16 committed:
Update storages.lua

omarcopires commented on data-otservbr-global/scripts/quests/oramond/the_ancient_sewers/actions_the_ancient_sewers.lua:
```diff
@@ -0,0 +1,83 @@
+local function revertItem(toPosition, getItemId, itemTransform)
```
```LUA
local config = {
	[21039] = { itemGerator = 21792, itemTransform = 21039 },
	[21040] = { itemGerator = 21792, itemTransform = 21040 },
}

local function revertItem(toPosition, getItemId, itemTransform)
	local tile = toPosition:getTile()
	if tile then
		local thing = tile:getItemById(getItemId)
		if thing then
			thing:transform(itemTransform)
		end
	end
end

local theAncientSewers = Action()

function theAncientSewers.onUse(player, item, fromPosition, target, toPosition, isHotkey)
	local gerator = config[item.itemid]
	if not gerator then
		return true
	end

	player:say((math.random(1, 100) < 50) and "<clong!> <clong!> There This piece fixed." or "<clong!> <clong!> <scrit scrit scrit>This should do it.", TALKTYPE_MONSTER_SAY)
	item:transform(gerator.itemGerator)
	addEvent(revertItem, 2 * 60 * 1000, toPosition, gerator.itemGerator, gerator.itemTransform)
	toPosition:sendMagicEffect(CONST_ME_BLOCKHIT)

	local currentMission = player:getStorageValue(Storage.Quest.U10_50.OramondQuest.TheAncientSewers.Mission)
	if currentMission >= 1 and currentMission < 20 then
		player:setStorageValue(Storage.Quest.U10_50.OramondQuest.TheAncientSewers.Mission, currentMission + 1)
	elseif currentMission == 20 then
		if player:getStorageValue(Storage.Quest.U10_50.DarkTrails.Mission03) < 1 then
			player:setStorageValue(Storage.Quest.U10_50.DarkTrails.Mission03, 1)
		end

		player:setStorageValue(Storage.Quest.U10_50.OramondQuest.TheAncientSewers.Mission, 21)
	end
	return true
end

for itemId, info in pairs(config) do
	theAncientSewers:id(itemId)
end

theAncientSewers:register()
```

htc16 replied:
I didn't understand?

htc16 replied:
yes, now I understand, sorry, I will update, thanks.

htc16 committed:
Oramond Quest - @marco

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Lions Rock Quest

htc16 committed:
Update storages.lua

htc16 committed:
Lions Rock Quest - minor tweaks to finish

htc16 committed:
Lion Rock Quest - Final Touch

htc16 committed:
Lion Rock Quest - Final Touch 2

htc16 committed:
Update gems.lua

htc16 committed:
Update gems.lua

omarcopires commented on data-otservbr-global/lib/core/constants.lua:
```diff
@@ -1,33 +0,0 @@
-ROSHAMUUL_MORTAR_THROWN = 20200
```
This file can be removed if it no longer contains any content.

htc16 replied:
Mesma coisa que comentei anteriormente.

omarcopires commented on data-otservbr-global/lib/core/storages.lua:
```diff
@@ -2392,51 +2235,261 @@ Storage = {
+					Temperature = 44311,
+					Kill = 44312,
+				},
+				Constants = {
```
I think the naming standard should be maintained and I also believe that they are not constant.

htc16 replied:
Eu importei elas do arquivo constantes, pois a missao em si acontece tudo dentro do constantes, sendo assim acho que fica mais viavel deixar dentro da storage para poder manipular os storages, dai os demais deverá ser usado quando para funcionar o quest log. Por isso mantive como constantes. Dai o arquivo constantes eu preferi deixar pois poderá ser usado em outro momento para outra finalidade ou os gerentes da comunidade pode por bem remove-lo se acharem que seria interessante.

omarcopires commented on data-otservbr-global/lib/core/storages.lua:
```diff
@@ -2392,51 +2235,261 @@ Storage = {
-			Roshamuul = {},
+			RoshamuulQuest = {
+				Constants = {
+					Roshamuul_Mortar_Thrown = 44401,
```
I think the naming standard should be maintained and I also believe that they are not constant.

htc16 replied:
Conforme dito anteriormente, apenas mantive como constante por causa da do arquivo de onde eu tirei eles.

omarcopires commented on data-otservbr-global/npc/gnomargery.lua:
```diff
@@ -148,15 +148,15 @@ local function creatureSayCallback(npc, creature, type, message)
 	////PARCEL DELIVERY////
```
These types of comments can also be removed, they don't help at all and make the code ugly.

htc16 replied:
Realmente, tenho tentado remover todos os comentarios que posso, mas alguns passam vou deixar aberto esse para eu lembrar de apagar.

omarcopires commented on data-otservbr-global/npc/gnombold.lua:
```diff
@@ -146,15 +146,15 @@ local function creatureSayCallback(npc, creature, type, message)
 	////GEOMANTIC CHARGES////
```
These types of comments can also be removed, they don't help at all and make the code ugly.

htc16 replied:
feito

omarcopires commented on data-otservbr-global/npc/gnome_trooper.lua:
```diff
@@ -56,23 +46,43 @@ npcType.onThink = function(npc, interval)
 	npcHandler:onThink(npc, interval)
 end
 
+if not DELIVERED_PARCELS then
```
```lua
if not DELIVERED_PARCELS then
	DELIVERED_PARCELS = {}
end

local response = {
	[0] = "It's a pipe! What can be more relaxing for a gnome than to smoke his pipe after a day of duty at the front. At least it's a chance to do something really dangerous after all!",
	[1] = "Ah, a letter from home! Oh - I had no idea she felt that way! This is most interesting!",
	[2] = "It's a model of the gnomebase Alpha! For self-assembly! With toothpicks...! Yeeaah...! I guess.",
	[3] = "A medal of honour! At last they saw my true worth!",
}

local function initializeParcelDelivery(player)
	local playerGuid = player:getGuid()
	if not DELIVERED_PARCELS[playerGuid] then
		DELIVERED_PARCELS[playerGuid] = {}
	end

	return DELIVERED_PARCELS[playerGuid]
end

local function greetCallback(npc, creature)
	local player = Player(creature)
	local playerGuid = player:getGuid()
	local deliveredParcels = initializeParcelDelivery(player)
	local parcelStatus = player:getStorageValue(Storage.Quest.U10_20.SpikeTaskQuest.Constants.Spike_Lower_Parcel_Main)
	
	if table.contains({-1, 4}, parcelStatus) or table.contains(deliveredParcels, npc:getId()) then
		return false
	end

	npcHandler:setMessage(MESSAGE_GREET, "Do you have something to deliver?")
	return true
end

local function creatureSayCallback(npc, creature, type, message)
	local player = Player(creature)
	local playerGuid = player:getGuid()
	local deliveredParcels = initializeParcelDelivery(player)
	local parcelStatus = player:getStorageValue(Storage.Quest.U10_20.SpikeTaskQuest.Constants.Spike_Lower_Parcel_Main)

	if MsgContains(message, "something") and not table.contains({-1, 4}, parcelStatus) then
		if table.contains(deliveredParcels, npc:getId()) then
			return true
		end

		if not player:removeItem(19219, 1) then
			npcHandler:say("But you don't have it...", npc, creature)
			return npcHandler:removeInteraction(npc, creature)
		end

		npcHandler:say(response[parcelStatus], npc, creature)
		player:setStorageValue(Storage.Quest.U10_20.SpikeTaskQuest.Constants.Spike_Lower_Parcel_Main, parcelStatus + 1)
		table.insert(deliveredParcels, npc:getId())
		npcHandler:removeInteraction(npc, creature)
	end
	return true
end
```

htc16 replied:
feito

htc16 committed:
The Cursed Crystal Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update movements_StepIn_CursedCrystal.lua

htc16 committed:
Merge branch 'Review-Quests-V10' of https://github.com/htc16/canary into Review-Quests-V10

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Storage organization

htc16 committed:
@omarcopires

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update gnome_trooper.lua

htc16 committed:
Merge branch 'Review-Quests-V10' of https://github.com/htc16/canary into Review-Quests-V10

htc16 committed:
More storage organization

htc16 committed:
More storage organization

htc16 committed:
Small adjustment to The Cursed Crystal quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Grimvale Quest

htc16 committed:
Nightmare Teddy Quest

htc16 committed:
The Great Dragon Hunt Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Lost Brother Quest

htc16 committed:
The Tainted Souls Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Ferumbras' Ascension Quest

Congratulations to the creator of this quest, it was the most complex so far and very well done.

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update storages.lua

htc16 committed:
Merge remote-tracking branch 'upstream/main' into Review-Quests-V10

dgprado-80 commented:
Npc Jamesfrancis from Heart of Destruction still buged. When you say hi to Jamesfrancis he doesn't respond.

htc16 commented:
> Npc Jamesfrancis from Heart of Destruction still buged. When you say hi to Jamesfrancis he doesn't respond.

This quest needs to be developed, I put it as a development need, did you see? This PR is only for reviewing the existing ones.

omarcopires commented on data-otservbr-global/npc/one_eyed_joe.lua:
```diff
@@ -56,88 +56,119 @@ npcType.onCloseChannel = function(npc, creature)
 
-local function creatureSayCallback(npc, creature, type, message)
-	local player = Player(creature)
+math.randomseed(os.time())
```
you could remove all others and leave only in global.lua to avoid problems.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_plagirath.lua:
```diff
@@ -0,0 +1,102 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_ragiaz.lua:
```diff
@@ -0,0 +1,114 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_rat_lever.lua:
```diff
@@ -0,0 +1,102 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_razzagorn.lua:
```diff
@@ -0,0 +1,102 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_shulgrax.lua:
```diff
@@ -0,0 +1,102 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_tarbaz.lua:
```diff
@@ -0,0 +1,102 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires commented on data-otservbr-global/scripts/quests/ferumbras_ascension/actions_zamulosh.lua:
```diff
@@ -0,0 +1,116 @@
+local config = {
```
this lever must be done with the new system, as well as the others in `scripts/actions/boss_levers`

htc16 replied:
I used a system that is complete, it checks the room beforehand, cleans the room, etc., the previous system didn't do that.

omarcopires replied:
In fact, the new system performs all the necessary checks. In addition to being more secure and avoiding code repetition.

htc16 committed:
Update one_eyed_joe.lua

htc16 committed:
Merge remote-tracking branch 'upstream/main' into Review-Quests-V10
